datamol/viz/_lasso_highlight.py

```diff
@@ -0,0 +1,439 @@
+
+    if search_molecules is None or (
+        isinstance(search_molecules, str) and len(search_molecules) == 0
+    ):
+        raise ValueError("Please enter valid search molecules or smarts")
+
+    # less than 1 throws File parsing error: PNG header not recognized over 5,000 leads to a DecompressionBombError later on
+    if canvas_width < 1 or canvas_width > 5000 or canvas_height < 1 or canvas_height > 5000:
+        raise ValueError(
+            "To avoid errors please choose a number between 1-5000 for the canvas width or height"
+        )
+
+    if isinstance(target_molecule, str):
+        target_molecule = dm.to_mol(target_molecule)
+
+    mol = prepare_mol_for_drawing(target_molecule, kekulize=True)
+    d = rdMolDraw2D.MolDraw2DCairo(500, 500)
+    d.drawOptions().updateAtomPalette({i: (0, 0, 0, 1) for i in range(100)})
+    d.DrawMolecule(mol)
+    d.ClearDrawing()
+
+    # get the atom indices for the search molecules
+    atom_idx_list = []
+    if isinstance(search_molecules, str):
+        smart_obj = Chem.MolFromSmarts(search_molecules)
+        matches = mol.GetSubstructMatches(smart_obj)
+        if not matches:
+            logger.warning(f"no matching substructure found for {search_molecules}")
+        else:
+            matched_atoms = set.union(*[set(x) for x in matches])
+            atom_idx_list.append(matched_atoms)
+
+    elif isinstance(search_molecules, dm.Mol):
+        matches = mol.GetSubstructMatches(search_molecules)
+        if not matches:
+            logger.warning(f"no matching substructure found for {dm.to_smiles(search_molecules)}")
+        else:
+            matched_atoms = set.union(*[set(x) for x in matches])
+            atom_idx_list.append(matched_atoms)
+
+    elif len(search_molecules) and isinstance(search_molecules[0], str):
+        for smart_str in search_molecules:
+            smart_obj = Chem.MolFromSmarts(smart_str)
+            matches = mol.GetSubstructMatches(smart_obj)
+            if not matches:
+                logger.warning(f"no matching substructure found for {smart_str}")
+            else:
+                matched_atoms = set.union(*[set(x) for x in matches])
+                atom_idx_list.append(matched_atoms)
+
+    elif len(search_molecules) and isinstance(search_molecules[0], dm.Mol):
+        for smart_obj in search_molecules:
+            matches = mol.GetSubstructMatches(smart_obj)
+            if not matches:
+                logger.warning(f"no matching substructure found for {dm.to_smiles(smart_obj)}")
+            else:
+                matched_atoms = set.union(*[set(x) for x in matches])
+                atom_idx_list.append(matched_atoms)
+
+    if len(atom_idx_list) == 0:
+        logger.warning("No matches found for the given search molecules")
+    else:
+        _draw_multi_matches(
+            d,
+            mol,
+            atom_idx_list,
+            r_min=0.3,
+            r_dist=0.12,
+            relative_bond_width=0.5,
```

Comment:
Allow the user to set `r_min`, r_dist`, `line_width` and `relative_bond_width` no?

Reply:
I choose to leave this out in hopes of making it easier to use.  Would this be better to have added in?

Reply:
I went ahead and added them. I think it's important to bring this kind of flexibility to the user (with good defaults).